Islandora/islandora_defaults
# Fix migration for resource type terms; Linked Agent field defaults to "No relation" option

**GitHub Issue**: Islandora/documentation#1531, Islandora/documentation#1770

* Other Relevant Links (Google Groups discussion, related pull requests,
 Release pull requests, etc.)
na

# What does this Pull Request do?
* Updates the default tags migration so that authority URIs will be loaded (Islandora/documentation#1531)
* Updates the field_linked_agent configuration to (1) include a "no relation" relation as the default; and (2) not include relator codes in the display values (Islandora/documentation#1770)

# What's new?
* Updates default tags migration so that the value of the `external_uri` field in the tags CSV will be ingested into `field_authority_link` if a vocabulary (such as Resource Types) uses this field instead of `field_authority_link`. This change was suggested by @kayakr [here](https://github.com/Islandora/documentation/issues/1531#issuecomment-819134857)
* Updates `field_linked_agent` config to...
  * make `relators:asn` with an empty string display value the first (i.e. de-facto default) value in the `rel_types` list
  * remove `relators:asn|Associated name` from the `rel_types` list because it caused a duplicate key error. 
  * remove relator codes and other parenthetical info from display values of all terms

* Does this change require documentation to be updated? 

I think it would be **best** to note in the documentation for this field that the default values that ship include the following deprecated terms: 

```
'relators:clb': 'Collaborator (clb; deprecated, use Contributor)'
'relators:grt': 'Graphic technician (grt; deprecated, use Artist)'
'relators:voc': 'Vocalist (voc; deprecated, use Singer)'
```
In stripping out the parenthetical info in the display values, we removed any indication of the fact that these terms are deprecated. Some institutions may want to remove them from the list or re-add the deprecated/use note, depending on their needs. 

* Does this change add any new dependencies? Yes, @seth-shaw-unlv is making some changes to the Typed Relation field type in `controlled_access_terms` that are related to the changes made to `field_linked_agent`

* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? Not sure/don't think so
* Could this change impact execution of existing code? Not sure/don't think so.

# How should this be tested?

Sorry these are going to be vague...

**the migration part (1531)**
I was not able to test this in our development environment because it is rebuilding from configs instead of actually starting from scratch and using the islandora_defaults code. Further, Nigel has disabled the migration from being picked up in our instances because it was causing a problem for something else (?), so I couldn't manually run it. 

I believe to test the migration part you will need to spin up a brand new Islandora completely from scratch. Or do something clever to get the migration config into an existing instance and manually run it. 

After the migration runs, examine a term in the Resource Types vocabulary and ensure the authority link from the tags CSV has been imported into the Authority Link field. 

**the field config part (1770)**
I was able to test this by doing `drush config:import --partial --source PATH/TO/ISLANDORA_DEFAULTS/CONFIG/INSTALL` on a branch in our dev environment

For me, this command got one error or warning unrelated to changes made here: apparently the field type of `field_description` changed at some point and now there's data in the field so it's a problem. If you see this, just ignore it. 

Steps to test: 

1. Create a new Repository Item and look at the Linked Agent field in the form. 
2. Verify that nothing is initially showing in the Relationship Type dropdown  (i.e. the default is now "no relation")
3. Choose a relationship type value from the list, verifying that it just says "Author" rather than "Author (aut)"

To test further, I think there is a dependency on further work @seth-shaw-unlv is doing on formatter/display for the Typed Relation field type in controlled_access_terms


# Additional Notes:
Not thinking of anything, other than next time I'm going to make two separate PRs for two separate issues instead of combining them like this. Seemed like it was going to be easier and Seth said it was fine, but it's bugging me :-) 

# Interested parties (not already tagged)
@mjordan @rosiel @dannylamb
